components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java

```diff
@@ -0,0 +1,233 @@
+            data.put(PERMISSIONS_FIELD, new JSONArray(permissionsList));
+        }
+        if (ArrayUtils.isNotEmpty(userList.toArray())) {
+            data.put(USERS_FIELD, new JSONArray(userList));
+        }
+        if (ArrayUtils.isNotEmpty(groupList.toArray())) {
+            data.put(GROUPS_FIELD, new JSONArray(groupList));
+        }
+        AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(getInitiatorId(),
+                LoggerUtils.getInitiatorType(getInitiatorId()), roleId, LoggerUtils.Target.Role.name(), ADD_ROLE_ACTION)
+                .data(jsonObjectToMap(data));
+        triggerAuditLogEvent(auditLogBuilder);
+    }
+
+    @Override
+    public void postUpdateRoleName(String roleId, String newRoleName, String tenantDomain) {
+
+        JSONObject data = new JSONObject();
+        data.put(ROLE_NAME_FIELD, newRoleName);
+        AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(getInitiatorId(),
+                LoggerUtils.getInitiatorType(getInitiatorId()), roleId, LoggerUtils.Target.Role.name(),
+                UPDATE_ROLE_NAME_ACTION).data(jsonObjectToMap(data));
+        triggerAuditLogEvent(auditLogBuilder);
+    }
+
+    @Override
+    public void postDeleteRole(String roleId, String tenantDomain) {
+
+        AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(getInitiatorId(),
+                LoggerUtils.getInitiatorType(getInitiatorId()), roleId, LoggerUtils.Target.Role.name(),
+                DELETE_ROLE_ACTION);
+        triggerAuditLogEvent(auditLogBuilder);
+    }
+
+    @Override
+    public void postGetUserListOfRole(List<UserBasicInfo> userBasicInfoList, String roleId, String tenantDomain) {
+
+        JSONObject data = new JSONObject();
+        if (ArrayUtils.isNotEmpty(userBasicInfoList.toArray())) {
+            ArrayList<String> usersList = new ArrayList<>();
+            for (UserBasicInfo user : userBasicInfoList) {
+                usersList.add(user.getId());
+            }
+            data.put(USERS_FIELD, new JSONArray(usersList));
+        }
+        AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(getInitiatorId(),
+                LoggerUtils.getInitiatorType(getInitiatorId()), roleId, LoggerUtils.Target.Role.name(),
+                GET_USERS_OF_ROLE_ACTION).data(jsonObjectToMap(data));
+        triggerAuditLogEvent(auditLogBuilder);
+    }
+
+    @Override
+    public void postUpdateUserListOfRole(String roleId, List<String> newUserIDList, List<String> deletedUserIDList,
+                                         String tenantDomain) {
+
+        JSONObject data = new JSONObject();
+        if (ArrayUtils.isNotEmpty(newUserIDList.toArray())) {
+            data.put(NEW_USERS_FIELD, new JSONArray(newUserIDList));
+        }
+        if (ArrayUtils.isNotEmpty(deletedUserIDList.toArray())) {
+            data.put(DELETED_USERS_FIELD, new JSONArray(deletedUserIDList));
+        }
+        if (isEnable()) {
```

Comment:
do we need this

Reply:
I checked that in certain cases the enabled checked is not respected. Hence, it is necessary to check if enabled at each event handling method. Hence, Will add 

```
if (!isEnable()) {
            return true;
        }
```

or 

```
if (!isEnable()) {
            return;
        }
```

at necessary places